app/project_type/git.py

```diff
@@ -154,7 +154,14 @@ def _fetch_project(self):
         self._git_remote_command_executor.execute(fetch_command, cwd=self._repo_directory)
 
         commit_hash = self._hash or 'FETCH_HEAD'
-        reset_command = 'git reset --hard {}'.format(commit_hash)
+
+        # git reset (--patch | -p) [<tree-ish>] [--] [<paths>...]
+        # For our use case, the '--' option acts as a delimiter to
+        # differentiate values that can be "tree-ish" or a "path"
+        # For example, the case where there is a file that has the same name as a
+        # branch.
+        # The <paths> value is optional and can be safely omitted.
+        reset_command = 'git reset --hard {} --'.format(commit_hash)
```

Comment:
Looks good. You could maybe abbreviate the comment; I think the "--" is a pretty common syntax for git commands for exactly this reason, so it probably doesn't need a huge explanation.

Should we also add the "--" to our git checkout command? Is there any other git command we do that we could add this to?